agent/http_test.go

```diff
@@ -122,6 +124,86 @@ func TestHTTPServer_UnixSocket_FileExists(t *testing.T) {
 	}
 }
 
+func TestHTTPServer_H2(t *testing.T) {
+	t.Parallel()
+
+	// Fire up an agent with TLS enabled.
+	a := &TestAgent{
+		Name:   t.Name(),
+		UseTLS: true,
+		HCL: `
+			key_file = "../test/client_certs/server.key"
+			cert_file = "../test/client_certs/server.crt"
+			ca_file = "../test/client_certs/rootca.crt"
+		`,
+	}
+	a.Start()
+	defer a.Shutdown()
+
+	// Make an HTTP/2-enabled client, using the API helpers to set
+	// up TLS to be as normal as possible for Consul.
+	tlscfg := &api.TLSConfig{
+		Address:  "consul.test",
+		KeyFile:  "../test/client_certs/client.key",
+		CertFile: "../test/client_certs/client.crt",
+		CAFile:   "../test/client_certs/rootca.crt",
+	}
+	tlsccfg, err := api.SetupTLSConfig(tlscfg)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	transport := api.DefaultConfig().Transport
+	transport.TLSClientConfig = tlsccfg
+	if err := http2.ConfigureTransport(transport); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	hc := &http.Client{
```

Comment:
Any reason not to wire up http2 transport when tls settings are provided in API client we create for the CLI [here](https://github.com/hashicorp/consul/blob/8a0195ee99557725bdc39870cec87a933c6a3c80/command/flags/http.go#L87)?

Reply:
I wanted to leave things in the default configuration for now and then flip the API client's default before the next major release, which will turn on HTTP/2 for anything using TLS.